app/services/update_wikidata_stats.rb

```diff
@@ -21,10 +46,117 @@ def import_summaries
 
   def update_wikidata_statistics
     return if course_revisions.empty?
-    stats = WikidataSummaryParser.analyze_revisions(course_revisions)
     crs_stat = CourseStat.find_by(course_id: @course.id) || CourseStat.create(course_id: @course.id)
-    crs_stat.stats_hash[wikidata.domain] = stats
+    # crs_stat.stats_hash[wikidata.domain] = stats
+    # crs_stat.save
+    stats_hash = crs_stat.stats_hash || {}
+
+    # Initialize arrays to store revisions with edit summaries and serialized stats
+    revisions_with_summary = []
+    revisions_with_serialized_stats = []
+
+    # Divide revisions based on edit summaries or serialized stats
+    course_revisions.each do |revision|
+      if revision.summary.present? && revision.summary.start_with?('{', '[')
```

Comment:
I think that knowledge about how to handle the `summary` field should be part of the Revision object.

I would recommend implementing a pair of methods for Revision, like `edit_summary` and `diff_stats`, and `edit_summary` would return an edit summary if that's what is present, and otherwise return `nil`, while `diff_stats` would return the already already-parsed stats if present and otherwise return `nil`.

I think this `starts_with` check is going to work for all edits to Wikidata Item namespace (as well as Lexeme, etc), but there's a chance it could fail for an edit in userspace or another conventional wiki namespace where edit summaries are not created automatically (so someone could make an edit that actually starts with one of these characters even though it's just a string and parsing it would through an error).